src/bokeh/plotting/contour.py

```diff
@@ -337,18 +346,20 @@ def _contour_coords(
         all_ys = []
         for level in levels:
             lines = cont_gen.lines(level)
+            if TYPE_CHECKING:
```

Comment:
Not sure I'd bother with these `TYPE_CHECKING` guards, an unconditional cast is fine with me. I also can't think of any other instances where we employ a `TYPE_CHECKING` guard outside the top of the file.

Reply:
Actually they are necessary in this specific situation. The type annotations are in a stub file `_contourpy.pyi` that shadows the compiled C++ module `_contourpy.whatever.so`. The stub file can be imported within a `TYPE_CHECKING` guard, and hence use of the imported types from that stub file also need to be within such a guard. If you import `_contourpy` outside of such a guard you don't get the type annotations. What I should do on the contourpy side is move these and similar type annotations out of the stub file and into a separate python file, but this will have to wait until the next release. In the meantime there are 2 options:

1. Keep this as it is.
2. Revert the type annotations which simply involves removing the last commit of this PR.

Reply:
is `cast("LineReturn_ChunkCombinedNan", lines)` an option here (I don't actually know if string annotations will work in this context or not). If not, I'd vote to just remove annotations for now.

Reply:
Yes, that works locally for me. I'll try it out on the full CI.